perceval/components/core_catalog/toffoli.py

```diff
@@ -0,0 +1,85 @@
+# MIT License
+#
+# Copyright (c) 2022 Quandela
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from deprecated import deprecated
+from perceval.components import Circuit, Unitary, Port
+from perceval.components.component_catalog import CatalogItem, AsType
+from perceval.utils import Encoding, Matrix, PostSelect
+
+
+class ToffoliItem(CatalogItem):
+    description = r"""Toffoli gate (CCNOT) with 6 heralded modes"""
+    str_repr = r"""                       ╭─────╮
+ctrl0 (dual rail) ─────┤     ├───── ctrl0 (dual rail)
+                  ─────┤     ├─────
+                       │     │
+ctrl1 (dual rail) ─────┤     ├───── ctrl1 (dual rail)
+                  ─────┤     ├─────
+                       │     │
+data (dual rail)  ─────┤     ├───── data (dual rail)
+                  ─────┤     ├─────
+                       ╰─────╯"""
+
+    def __init__(self):
+        super().__init__("toffoli")
+        self._default_opts['type'] = AsType.PROCESSOR
+
+    @deprecated(version="0.10.0", reason="Use build_circuit or build_processor instead")
```

Comment:
Don't bother implementing deprecated methods in new classes. I removed the abstractmethod modifier so that we can get rid of it in new components (see mzi.py for instance)